xdsl/pattern_rewriter.py

```diff
@@ -92,60 +92,47 @@ class PatternRewriter(PatternRewriterListener):
     has_done_action: bool = field(default=False, init=False)
     """Has the rewriter done any action during the current match."""
 
+    def insert_op_at_location(
+        self, op: Operation | Sequence[Operation], insertion_point: InsertPoint
+    ):
+        """Insert operations at a certain location in a block."""
+        self.has_done_action = True
+        op = [op] if isinstance(op, Operation) else op
+        if not op:
+            return
+        for op_ in op:
+            Builder(insertion_point).insert(op_)
```

Comment:
Can we hoist the builder construction out of the loop? I'm not deeply familiar with it's API, but this seems a bit wasteful imo.

Reply:
Since putting this up I've realised this is not the right approach, we should work with the rewriter directly